merlin/models/tf/blocks/core/masking.py

```diff
@@ -328,18 +339,12 @@ def compute_feature_mask(self, item_ids: tf.Tensor, training: bool = False) -> t
             )
             self.labels.scatter_nd_update(indices=indices, updates=tf.gather_nd(item_ids, indices))
             mask_labels = self.labels != self.padding_idx
+            mask_inputs = mask_labels
         else:
-            labels = item_ids[:, 1:]
-            labels = tf.concat(
-                [
-                    labels,
-                    tf.zeros((tf.shape(labels)[0], 1), dtype=labels.dtype),
-                ],
-                axis=-1,
-            )
-            mask_labels = labels != self.padding_idx
+            mask_labels = item_ids != self.padding_idx
```

Comment:
I guess we will still need to shift the sequence of item ids as the first element won't be used as `target` (we don't have any prior information to feed to the model and getting predictions)

Reply:
Hey Sara. Yes, shifting labels for next-click prediction with CLM is still necessary. I just removed the [shifting](https://github.com/NVIDIA-Merlin/models/pull/464/files/9a73b2fbef8bbfcd5ac5e70f4498b745b6794cc3#diff-b5dafb33a21eb7e088c3576769b9bc2c184742a8f5096dbad501304786a22f1eL187-L193) here because it was not necessary, as this method returns the label mask and not the label values

Reply:
Thanks for the answer! I guess it would impact the definition of targets [here](https://github.com/NVIDIA-Merlin/models/blob/6fc41b65e26c50a336053258ba602293d8a9e9d3/merlin/models/tf/blocks/core/masking.py#L379). As the current implementation of MaskingHead uses the masking tensor stored in the context to retrieve targets?